flycheck.el

```diff
@@ -7744,7 +7744,12 @@ Otherwise return the previously used cache directory."
 Look up the directory hierarchy from FILE for a directory
 containing a file that matches REGEXP."
   (locate-dominating-file
-   file
+   ;; We must not pass regular file here because
+   ;; `locate-dominating-file' will just pass it into it's second
+   ;; lambdaargument . But our lambda expects only directories.
+   (if (file-directory-p file)
```

Comment:
That's a fair point.  It does make the function more robust, even though we shouldn't run into any issue considering we always pass a filename at present.